profiles/pp/modules/custom/idevels_twitter_login/src/Controller/IdevelsTwitterLoginController.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+/**
+ * @file
+ * Contains IdevelsTwitterLoginController.
+ */
+
+namespace Drupal\idevels_twitter_login\Controller;
+
+use Abraham\TwitterOAuth\TwitterOAuth;
+use Drupal\Core\Controller\ControllerBase;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Session\Session;
+
+/**
+ * Site signup/login using Twitter.
+ */
+class IdevelsTwitterLoginController extends ControllerBase {
+
+  /**
+   * Log in using Twitter. This is a single action for all auth flow.
+   *
+   * @return \Symfony\Component\HttpFoundation\Redirect
+   *   The auth page.
+   *
+   * @throws \Abraham\TwitterOAuth\TwitterOAuthException
+   *   Twitter API error.
+   */
+  public function unifiedLoginRegister() {
+    $idevels_twitter_login_config = $this->config('idevels_twitter.system');
+    $consumer_key = $idevels_twitter_login_config->get('idevels_twitter_consumer_key');
+    $secret_key = $idevels_twitter_login_config->get('idevels_twitter_consumer_secret');
+    if (!$consumer_key || strlen($consumer_key) == 0) {
+      drupal_set_message(t('Missing Twitter consumer key. Please contact site administrator.'), 'error');
+      return (new RedirectResponse(\Drupal::url('<front>')));
+    }
+    if (!$secret_key|| strlen($consumer_key) == 0) {
+      drupal_set_message(t('Missing Twitter secret key. Please contact site administrator.'), 'error');
+      return (new RedirectResponse(\Drupal::url('<front>')));
+    }
+    libraries_load('twitteroauth');
+
+    $session = new Session();
```

Comment:
there's session manager for that in d8, follow https://www.drupal.org/node/2228393 and its meta